src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -0,0 +1,172 @@
+
+import { useBackgroundBlur } from '../../../providers/BackgroundBlurProvider';
+import { useVideoInputs } from '../../../providers/DevicesProvider';
+import { useLocalVideo } from '../../../providers/LocalVideoProvider';
+import { useMeetingManager } from '../../../providers/MeetingProvider';
+import { DeviceConfig } from '../../../types';
+import {
+  isOptionActive,
+  videoInputSelectionToDevice,
+} from '../../../utils/device-utils';
+import { ControlBarButton } from '../../ui/ControlBar/ControlBarButton';
+import { Camera, Spinner } from '../../ui/icons';
+import PopOverItem from '../../ui/PopOver/PopOverItem';
+import PopOverSeparator from '../../ui/PopOver/PopOverSeparator';
+
+interface Props {
+  /** The label that will be shown for video input control, it defaults to `Video`. */
+  label?: string;
+  /** The label that will be shown when the Background Blur checkbox is checked */
+  backgroundBlurOnLabel?: string;
+  /** The label that will be shown when the Background Blur checkbox is not checked */
+  backgroundBlurOffLabel?: string;
+}
+
+const videoInputConfig: DeviceConfig = {
+  additionalDevices: true,
+};
+
+const VideoInputBackgroundBlurControl: React.FC<Props> = ({
+  label = 'Video',
+  backgroundBlurOnLabel = 'Background Blur Enabled',
+  backgroundBlurOffLabel = 'Enable Background Blur',
+}) => {
+  const meetingManager = useMeetingManager();
+  const { devices, selectedDevice } = useVideoInputs(videoInputConfig);
+  const { isVideoEnabled, toggleVideo } = useLocalVideo();
+  const { isBackgroundBlurSupported, createBackgroundBlurDevice } =
+    useBackgroundBlur();
+  const [isVideoTransformCheckBoxOn, setisVideoTransformCheckBoxOn] =
+    useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [
+    dropdownWithVideoTransformOptions,
+    setDropdownWithVideoTransformOptions,
+  ] = useState<ReactNode[] | null>(null);
+  const [activeVideoDevice, setDevice] = useState<
+    Device | VideoTransformDevice | null
+  >(meetingManager.selectedVideoInputDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  useEffect(() => {
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    const deviceOptions: ReactNode[] = devices.map((option) => (
+      <PopOverItem
+        key={option.deviceId}
+        children={<span>{option.label}</span>}
+        checked={isOptionActive(selectedDevice, option.deviceId)}
+        onClick={async (): Promise<void> => {
+          // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+          // If background blur is not on, then do a normal video selection
+          if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
```

Comment:
Can we move this logic to meetingManager or somewhere else that is not in the UI?

Reply:
No i meant move this entire code block to meetingManager (since all our device selection logic is in there). Ideally we should just need a selectTransformDevice hook which you can call it here.

Reply:
I dont think we can do it here, since the `activeVideoDevice` here is managed by this component.

Reply:
We can pass it as input down right?